tests/activation/test_xonsh_activation.py

```diff
@@ -0,0 +1,57 @@
+import os
+import subprocess
+import sys
+from os.path import join, normcase
+
+import pytest
+
+import virtualenv
+from tests.lib import need_executable
+
+XONSH_COMMAND = "xonsh.exe" if virtualenv.is_win else "xonsh"
+
+
+def need_xonsh(fn):
+    return pytest.mark.xonsh(need_executable("xonsh", (XONSH_COMMAND, "--version"))(fn))
+
+
+def print_python_exe_path():
+    return "{} -c 'import sys; print(sys.executable)'".format(virtualenv.expected_exe)
+
+
+@need_xonsh
+def test_activate_with_xonsh(tmpdir, monkeypatch):
+    monkeypatch.chdir(tmpdir)
+    home_dir, _, __, bin_dir = virtualenv.path_locations(str(tmpdir.join("env")))
+    virtualenv.create_environment(home_dir, no_pip=True, no_setuptools=True, no_wheel=True)
```

Comment:
how is this test passing under python 2 🤔 if xonosh cannot run with it

Reply:
The test seems to be appropriately skipped on linux, windows, and mac on Python 2.7.  Are you seeing something different?

Reply:
Oh I understand, it's not installed on python 2.7 so it's skipped. This works by chance as https://github.com/pypa/virtualenv/blob/master/tests/lib/__init__.py#L14 should mean we never skip inside the CI 😮 please add some logic inside the needs_xonosh to always skip if xonosh is not installed.

Reply:
Ok! Does it look good now?